demux/demux_lavf.c

```diff
@@ -736,6 +769,72 @@ static void handle_new_stream(demuxer_t *demuxer, int i)
     select_tracks(demuxer, i);
 }
 
+static void detect_program_changes(demuxer_t *demuxer)
+{
+    lavf_priv_t *priv = demuxer->priv;
+    struct sh_stream *old_stream = NULL, *new_stream = NULL;
+    AVProgram *program;
+    int i, j;
+
+    if (priv->avfc->nb_programs < 1)
+        return;
+    if (priv->program.num == 0)
+        return;
+
+    for (i = 0; i < priv->avfc->nb_programs; i++)
+        if (priv->avfc->programs[i]->id == priv->program.num)
+            break;
```

Comment:
Unlike in FFmpeg, we also declare variables where we need them, like in the loop body. Also, mpv code tries to avoid a loop variable used in a `for` outside of the `for` statement.